tle/cogs/handles.py

```diff
@@ -830,6 +865,145 @@ async def role(self, ctx, action: str, which: str):
+            await ctx.send(
+                f"{trusted_role.mention} role granted to {target_user.mention} by {ctx.author.mention}."
+            )
+        except discord.Forbidden:
+            raise HandleCogError(f"No permissions to assign the '{trusted_role_name}' role.")
+        except discord.HTTPException as e:
+            raise HandleCogError(f"Failed to assign the role due to an unexpected error: {e}")
+        
+    @handle.command(brief='Grant Trusted role to old members without Purgatory role.')
+    @commands.has_role(constants.TLE_ADMIN)
+    async def grandfather(self, ctx):
+        """Grants the Trusted role to all members who joined before April 21, 2025,
+        and do not currently have the Purgatory role.
+        """
+        guild = ctx.guild
+        trusted_role_name = constants.TLE_TRUSTED
+        purgatory_role_name = 'Purgatory'
+
+        trusted_role = discord.utils.get(guild.roles, name=trusted_role_name)
+        if trusted_role is None:
+            raise HandleCogError(f"The role '{trusted_role_name}' does not exist in this server.")
+
+        purgatory_role = discord.utils.get(guild.roles, name=purgatory_role_name)
+        # If Purgatory role doesn't exist, we assume no one has it.
+        if purgatory_role is None:
+            self.logger.warning(f"Role '{purgatory_role_name}' not found in guild {guild.name} ({guild.id}). Proceeding without Purgatory check.")
+
+        # Cutoff date: April 21, 2025, 00:00:00 UTC
+        cutoff_date = dt.datetime(2025, 4, 21, 0, 0, 0, tzinfo=dt.timezone.utc)
+
+        added_count = 0
+        skipped_purgatory = 0
+        skipped_already_trusted = 0
+        skipped_join_date = 0
+        processed_count = 0
+
+        status_message = await ctx.send(f"Processing members for grandfathering Trusted...")
+
+        members_to_process = list(guild.members) # Create a list to avoid issues if members leave/join during processing
+        total_members = len(members_to_process)
+
+        for i, member in enumerate(members_to_process):
+            processed_count += 1
+            # Update status periodically
+            if i % 100 == 0 and i > 0:
+                await status_message.edit(content=f"Processing members... ({i}/{total_members})")
+
+            # Check join date
+            if member.joined_at is None:
+                # Cannot determine join date, skip
+                skipped_join_date += 1
+                continue
+
+            # Make member.joined_at timezone-aware (assuming it's UTC, which discord.py uses)
+            member_joined_at_aware = member.joined_at.replace(tzinfo=dt.timezone.utc)
+
+            if member_joined_at_aware >= cutoff_date:
+                skipped_join_date += 1
+                continue
+
+            # Check if user already has the Trusted role
+            if trusted_role in member.roles:
+                skipped_already_trusted += 1
+                continue
```

Comment:
Again, don't describe the what but the why. E.g. like above
```suggestion
            if trusted_role in member.roles:
                # User already trusted, skip
                skipped_already_trusted += 1
                continue
```